### PR TITLE
Swaps V2 Integration

### PR DESCRIPTION
**Pull request by meppsilon**
**Description**

There are five changes in this integration:
1. Incorporate swaps-controller changes (which change the swaps api url to v2)
2. Change all occurrences of "occurrences" to "occurrences" to align with the rename in the v2 API
3. Allow polygon network for swaps
4. Enable trades with direct wrapping
5. Provide text for new quote source: contract

~~**Depends on:** https://github.com/MetaMask/swaps-controller/pull/66~~

Minimal QA requirements:
1. Perform swaps on Mainnet, Binance, Polygon
2. Perform swaps that wrap and unwrap the native token on all networks

### CODE REVIEW

**Comment by wachunei** [comment]
This LGTM, however since I was involved I'd wait for a review from @MetaMask/mobile-devs

File: package.json [hunk 1]
```diff
@@ -80,7 +80,7 @@
     "@metamask/contract-metadata": "^1.23.0",
     "@metamask/controllers": "^14.0.2",
     "@metamask/etherscan-link": "^2.0.0",
-    "@metamask/swaps-controller": "^5.0.0",
+    "@metamask/swaps-controller": "file:./metamask-swaps-controller-v6.0.0-129faee.tgz",
```
Comment (rickycodes) [inline package.json:83]: do we not want to publish these before merge?

Comment (wachunei) [reply package.json:83]: Yes, we will definitely, very 🔜

Comment (wachunei) [reply package.json]: updated ☝️

**Review by rickycodes — APPROVED** [review]
LGTM

**Comment by wachunei** [comment]
Converting to draft because of remaining tasks, check description for the list

**Review by cortisiko — APPROVED** [review]
This guy is 👍

**Pull request merged**